lesson17/src/js/main.js

```diff
@@ -0,0 +1,225 @@
+  loadingImage.src = "images/loading.gif";
+  loadingImage.width = 80;
+  loadingImage.height = 80;
+  loadingImage.classList.add("loading__image");
+  fragment.appendChild(renderLoadingBox).appendChild(loadingImage);
+  document.body.insertBefore(fragment, main);
+}
+
+function removeLoading() {
+  document.getElementById("js-loading").remove();
+}
+
+function displayErrorMassage(error) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+  const createTextBox = document.createElement("p");
+  createTextBox.classList.add("");
+  createTextBox.textContent = error;
+  slideImageArea.appendChild(createTextBox);
+}
+
+function renderSlideArea() {
+  const slideImageArea = document.createElement("div");
+  slideImageArea.classList.add("mainvisual__images__inner");
+
+  const sliderArea = document.querySelector(".mainvisual__images");
+
+  sliderArea.appendChild(slideImageArea);
+}
+
+function renderSlideImage(item) {
+  const slideImageArea = document.querySelector(".mainvisual__images__inner");
+
+  const slider = document.createElement("ul");
+  slider.classList.add("slider");
+
+  const fragment = document.createDocumentFragment();
+  item.forEach((image, i) => {
+    const sliderItem = document.createElement("li");
+    sliderItem.classList.add("slider__item");
+    sliderItem.setAttribute("data-slide-index", i + 1);
+
+    const sliderImage = document.createElement("img");
+    sliderImage.classList.add("slider__image");
+    sliderImage.src = image.image;
+    sliderImage.width = image.width;
+    sliderImage.height = image.height;
+
+    if (i === 0) {
+      sliderItem.classList.add("is-active");
+    }
+
+    fragment.appendChild(sliderItem).appendChild(sliderImage);
+  });
+
+  slideImageArea.appendChild(slider).appendChild(fragment);
+}
+
+function renderSlideButton() {
+  let fragment = document.createDocumentFragment();
+
+  const buttonArea = document.createElement("div");
+  buttonArea.classList.add("slider__button");
+
+  const prevButton = document.createElement("button");
+  prevButton.setAttribute("class", "arrow prev");
+  prevButton.id = "js-button_prev";
+  prevButton.setAttribute("aria-label", "previous");
+  prevButton.textContent = "previous";
+
+  fragment.appendChild(prevButton);
+
+  const nextButton = document.createElement("button");
+  nextButton.setAttribute("class", "arrow next");
+  nextButton.id = "js-button_next";
+  nextButton.textContent = "next";
+  nextButton.setAttribute("aria-label", "next");
+
+  fragment.appendChild(nextButton);
+
+  const getSlideArea = document.querySelector(".mainvisual__images");
+
+  getSlideArea.appendChild(buttonArea).appendChild(fragment);
+}
+
+function renderPagenation(item) {
+  const pagenation = document.createElement("div");
+  pagenation.classList.add("slider__pagination");
+
+  let fragment = document.createDocumentFragment();
+
+  const current = document.createElement("span");
+  current.classList.add("current");
```

Comment:
IMO
From what I have seen, it is only operated by JS, so I feel that 'ID' with js-prefix is fine.
The same goes for subsequent 'separation' and 'total'.

Reply:
Certainly current and total do not specify the style, so I changed it to id.

seperation had a style,
I don't operate it with js at all, so I left it as a class.

[3bebf77](https://github.com/nyattsu72/hanson-frontend-cource/pull/25/commits/3bebf77ebd02fa7bcd742758e12a9786b7d0bc20)